src/backend/distributed/executor/executor.c

```diff
@@ -1844,14 +1902,19 @@ PlacementExecutionDone(TaskPlacementExecution *placementExecution, bool succeede
 		executionOrder == EXECUTION_ORDER_SEQUENTIAL)
 	{
 		TaskPlacementExecution *nextPlacementExecution = NULL;
-		int placementExecutionCount PG_USED_FOR_ASSERTS_ONLY =
-			shardCommandExecution->placementExecutionCount;
+		int placementExecutionCount = shardCommandExecution->placementExecutionCount;
 
 		/* find a placement execution that is not yet marked as failed */
 		do {
 			int nextPlacementExecutionIndex =
 				placementExecution->placementExecutionIndex + 1;
```

Comment:
I'm a little puzzled by this line, since I would expect `nextPlacementExecutionIndex` to keep incrementing as we iterate the loop.